src/main/scala/centaur/test/Test.scala

```diff
@@ -100,28 +98,37 @@ object Operations {
   }
 
   def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Test[Unit] = {
-    val consistencyTimeout = Future { Thread.sleep(CentaurConfig.metadataConsistencyTimeout.toMillis) }
+
+    def eventually(startTime: OffsetDateTime, timeout: FiniteDuration)(f: => Try[Unit]): Try[Unit] = {
```

Comment:
It just needs to be stack friendly. My original one was just basic recursive so it was not.